parsers/nmparser/structs.go

```diff
@@ -127,26 +129,38 @@ type ExtData struct {
 	EstimationLines   [][]string
 }
 
-// Status supports extended states beyond true and false
-type Status int
+// HeuristicStatus supports extended states beyond true and false
+type HeuristicStatus int
 
 const (
-	// Undefined Status, default value, not set to true or false
-	Undefined Status = iota
-	// True Status, explicitly set to true
-	True
-	// False Status, explicitly set to true
-	False
+	// HeuristicUndefined default value, not set to true or false
+	HeuristicUndefined HeuristicStatus = iota
+	// HeuristicTrue explicitly set to true
+	HeuristicTrue
+	// HeuristicFalse explicitly set to true
+	HeuristicFalse
 )
 
-func (s Status) String() string {
-	return [...]string{"Undefined", "True", "False"}[s]
+func (s HeuristicStatus) String() string {
+	return toString[s]
+}
+
+var toString = map[HeuristicStatus]string{
+	HeuristicUndefined: "HeuristicUndefined",
+	HeuristicTrue:      "HeuristicTrue",
+	HeuristicFalse:     "HeuristicFalse",
 }
 
-// // ToBool convert status to bool
-// func (s *Status) ToBool() bool {
-// 	if *s == True {
-// 		return true
-// 	}
-// 	return false
-// }
+var toID = map[string]HeuristicStatus{
+	"HeuristicUndefined": HeuristicUndefined,
+	"HeuristicTrue":      HeuristicTrue,
+	"HeuristicFalse":     HeuristicFalse,
+}
+
+// MarshalJSON marshals the enum as a quoted json string
```

Comment:
we will need to add an unmarshal json as well I believe, especially for our tests that will use golden json files as the source of truth